### PR TITLE
Feature/hadasfa/skeleton

### PR DESCRIPTION
**Pull request by hadasfa**
**Description: Add support on circle, text and rectangle skeletons**
https://monday.monday.com/boards/245345663/pulses/1436820196

https://user-images.githubusercontent.com/72390374/124572285-35b98500-de51-11eb-9710-dc1d369a1cce.mov

#### Basic
- [ ] Used plop (npm run plop) to create a new component. This command will create a folder for your component and will update index.js file
- [V ] PR has description
- [V] New component is functional and using React Hooks. 
- [ ] New React Hooks were added to /src/hooks folder (optional)
- [V] Component defines PropTypes
- [] There is Readme for the component, which explains the main aspects of how to use it
#### Style
- [ V] Styles are added to NewComponent.scss file inside of the NewComponent folder
- [V] CSS selectors are named using [BEM convention](http://getbem.com/naming/) 
- [V] Design is compatible with [Monday Design System](https://design.monday.com/)
#### Storybook
- [V] Stories were added to /src/NewComponent/__stories__/NewComponent.stories.js file
- [V] Stories include sandbox story with knobs for each property
- [V] Stories include all flows of using the component
- [V] Component passed Accessibility Plugin checks
#### Tests
- [ V ] Tests are compliant with TESTING_README.md instructions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "monday-ui-react-core",
-  "version": "0.3.49",
+  "version": "0.3.51",
```
Comment (hadasfa) [inline package.json:3]: still need to figure out:
- shimmer animation
- a11y - how to handle skeletons?
- all grey color key (replace for black mode)

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "monday-ui-react-core",
-  "version": "0.3.50",
+  "version": "0.3.51",
```
Comment (hadasfa) [inline package.json:3]: need to check:
-best way to attribute a11y (support role/aria label?)
-better animation for shimmer than margin?

File: src/components/Skeleton/SkeletonConstants.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import PropTypes from "prop-types";
+
+// Fixed names
+export const SKELETON_CIRCLE_TYPE = "circle";
+export const SKELETON_TEXT_TYPE = "text";
+export const SKELETON_RECTANGLE_TYPE = "rectangle";
+export const SKELETON_COSTUME_SIZE = "costume";
```
Comment (MosheZemah) [inline src/components/Skeleton/SkeletonConstants.jsx:7]: custom
costume is for Purim :)

File: src/components/Skeleton/Skeleton.scss [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    height: $circle-default-size;
+    width: $circle-default-size;
+    @include skeleton($circle-radius);
+  }
+
+  &_text {
+    width: 162px;
+    @include skeleton;
+
+    &--h1, &--h2, &--h3 {
```
Comment (MosheZemah) [inline src/components/Skeleton/Skeleton.scss:67]: maybe you should use styles/typography.scss i.e. - .element-type-h1

**Review by MosheZemah — CHANGES_REQUESTED** [review]

File: src/components/Skeleton/SkeletonConstants.jsx (hunk 3 above)
Comment (hadasfa) [reply src/components/Skeleton/SkeletonConstants.jsx:7]: oops ^^"

File: src/components/Skeleton/Skeleton.scss (hunk 4 above)
Comment (hadasfa) [reply src/components/Skeleton/Skeleton.scss]: the size of the div elements are not identical to the given sizes. insted I add sass consts for the text skeleton heights in typography.scss

**Review by MosheZemah — APPROVED** [review]

**Pull request merged**